security/src/main/java/org/apache/geode/examples/security/Example.java

```diff
@@ -0,0 +1,156 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.examples.security;
+
+import static org.apache.geode.distributed.ConfigurationProperties.SECURITY_CLIENT_AUTH_INIT;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.client.ServerOperationException;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.security.NotAuthorizedException;
+
+public class Example {
+  private static Logger logger = LogService.getLogger();
+  private static final String REGION1 = "region1";
+  private static final String REGION2 = "region2";
+
+  // We use the data present in the loader example
+  private static final String[] AUTHORS =
+      ("Anton Chekhov,C. J. Cherryh,Dorothy Parker,Douglas Adams,Emily Dickinson,"
+          + "Ernest Hemingway,F. Scott Fitzgerald,Henry David Thoreau,Henry Wadsworth Longfellow,"
+          + "Herman Melville,Jean-Paul Sartre,Mark Twain,Orson Scott Card,Ray Bradbury,Robert Benchley,"
+          + "Somerset Maugham,Stephen King,Terry Pratchett,Ursula K. Le Guin,William Faulkner")
+              .split(",");
+
+  private String username;
+  private String password;
+  private final ClientCache cache;
+
+
+  private Example(String username, String password) {
+    this.username = username;
+    this.password = password;
+    Properties props = new Properties();
+    props.setProperty(ExampleAuthInit.USER_NAME, this.username);
+    props.setProperty(ExampleAuthInit.PASSWORD, this.password);
+    props.setProperty(SECURITY_CLIENT_AUTH_INIT, ExampleAuthInit.class.getName());
+    // connect to the locator using default port 10334
+    cache = new ClientCacheFactory(props).setPoolSubscriptionEnabled(true)
+        .addPoolLocator("localhost", 10334).create();
+  }
+
+  /** A far-from-comprehensive example of integrated security in action. */
+  public static void main(String[] args) {
+    printCurrentData();
+
+    getDataExample("dataReader", false);
+    getDataExample("dataWriter", true);
+
+    copyDataExample("dataWriter", true);
+    copyDataExample("dataReader", true);
+    copyDataExample("dataAdmin", false);
+
+    printCurrentData();
+  }
+
+  /** Prints the data currently present in both /region1 and /region2 */
+  private static void printCurrentData() {
```

Comment:
Does printing this data have any side effects related to security?  If not I think we should remove it.